### PR TITLE
Add json-api-client to this repo

### PR DESCRIPTION
**Pull request by eatyourgreens**
Move the `json-api-client` code into `lib/json-api-client` and require it from there. Remove `json-api-client` as a dependency.

### CODE REVIEW

**Comment by eatyourgreens** [comment]
If we merge this, we can deprecate https://github.com/zooniverse/json-api-client.

https://github.com/zooniverse/Panoptes-Front-End/pull/6369 is a feature branch that's set up to test this version of the client.

**Comment by eatyourgreens** [comment]
The JSON-API client code here is from v7.0.0.

File: package.json [hunk 1]
```diff
@@ -7,8 +7,10 @@
   "license": "Apache-2.0",
   "repository": "zooniverse/panoptes-javascript-client",
   "dependencies": {
-    "json-api-client": "^7.0.2",
-    "local-storage": "^2.0.0"
+    "normalizeurl": "~1.0.0",
+    "local-storage": "^2.0.0",
+    "sugar-client": "^2.0.0",
```
Comment (mcbouslog) [inline package.json:12]: SugarClient is part of of PJC now?

Comment (eatyourgreens) [reply package.json:12]: Good point. It wasn’t when I opened this PR, and this branch needs updating.

**Review by mcbouslog — APPROVED** [review]
Tested this at https://pr-6369.pfe-preview.zooniverse.org/ throughly. Signed in/out, made classifications, posted/edited to Talk, created a project, edited a project/workflow/task. I think this is working well, but let me know if there are any user actions I've neglected to test.

So I think this is good to go! 👍

**Pull request merged**